Classes/Domain/Model/DateTime.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace SMS\FluidComponents\Domain\Model;
+
+use Exception;
+use SMS\FluidComponents\Interfaces\ConstructibleFromDateTime;
+use SMS\FluidComponents\Interfaces\ConstructibleFromInteger;
+use SMS\FluidComponents\Interfaces\ConstructibleFromString;
+
+class DateTime extends \DateTime implements ConstructibleFromString, ConstructibleFromInteger, ConstructibleFromDateTime
+{
+    /**
+     * Convert string input to datetime object
+     *
+     * @param string $value
+     * @return DateTime|false
+     * @throws Exception
+     */
+    public static function fromString(string $value)
+    {
+        return new static($value);
+    }
+
+    /**
+     * Convert UNIX timestamp to datetime object
+     *
+     * @param int $value
+     * @return self
+     * @throws Exception
+     */
+    public static function fromInteger(int $value): self
+    {
+        return new static('@' . $value);
+    }
+
+    /**
+     * Passes datetime object
+     *
+     * @param \DateTime $value
+     * @return static
+     * @throws Exception
+     */
+    public static function fromDateTime(\DateTime $value): self
+    {
+        return new static('@' . $value->getTimestamp());
```

Comment:
It's probably better to use a formatted date here instead of the timestamp because of microseconds and timezones. Something like this could work:
```php
return new static($value->format(\DateTimeInterface::RFC3339_EXTENDED);
```